### ISSUE TITLE
Lint PHP code examples

### ISSUE THREAD

**Issue by Nyholm** [symfony/symfony-docs#12184]
When correcting a [typo with missing semi colons](https://github.com/symfony/symfony-docs/pull/12183), one thing got me thinking. Could we use github action to make sure all PHP code examples have the correct syntax?

We could also extend this to yaml and xml. 

Something similar to @OskarStark's https://github.com/OskarStark/php-cs-fixer-ga

**Comment by OskarStark** [comment]
Hey Tobias 👋 

I already tried this:
* extract all code examples
* lint them with php-cs-fixer

but it doesn't work as expected.

Lets assume we have the following example:
```php
use Symofny \Component\Foo\Bar;

$bar = new Bar();
```
It looks valid, but the class `Bar` does not exist under the given namespace. Therefore we we need to use an exact version of the component (think of all the versionadded directives).
Maybe we could get this working, but we need alle external vendors like doctrine, Twig etc., too 👀 

Maybe we can fix such simple semicolon issues or sth. else.

I will try to dive into this topic when I find some time. :)

**Comment by Nyholm** [comment]
Yeah. We cannot run phpstan or something that does a deep analysis of the code. I think php-cs-fixer may be too much. But just something that verifies the syntax.

**Comment by wouterj** [comment]
Yeah, this is one item that has been on my wish list for years. I think extracting the code examples and running them with https://www.php.net/php-check-syntax would be a good thing. We have to update some code examples then (because we not always use `...` inside comments for instance), but I don't think that has a negative impact on the code examples.

**Comment by javiereguiluz** [comment]
Thanks for proposing this. I agree that if we had something with zero maintenance and which worked perfectly for any code, that'd be an awesome addition to the docs.

However, if we can't do that, I don't think it's a big loss. Doc contributions (from the Docs Team or the community) very rarely contain syntax mistakes and if they do, they are quickly fixed by the community reviews. But this is very rare; I can't remember merging PRs to fix syntax errors.
